cfme/containers/topology.py

```diff
@@ -1,23 +1,33 @@
 # -*- coding: utf-8 -*-
+import attr
+
 from navmazing import NavigateToAttribute
 
+from cfme.modeling.base import BaseCollection, BaseEntity
 from cfme.common import TopologyMixin
 from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
-from cfme.utils.appliance import Navigatable
 
 
-class Topology(TopologyMixin, Navigatable):
+class Topology(TopologyMixin, BaseEntity):
 
-    def __init__(self, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
+    region = attr.ib(default=None)
+    provider = attr.ib(default=None)
+    datastore = attr.ib(default=None)
 
     @classmethod
     def load_topology_page(cls):
         navigate_to(cls, 'All')
 
 
+@attr.s
+class TopologyCollection(BaseCollection):
+    def __init__(self, appliance=None):
```

Comment:
@amritrajck, Please fallow collection-v3 guidelines.

Reply:
Could you please explain in this context? Do you mean there shouldn't be an init  and should leave the class currently empty?

Reply:
@amritrajck fallow this [collection design](http://cfme-tests.readthedocs.io/en/master/guides/collections.html) for better understanding.
Actully, respective `__init__()`  method is hidden inside the `BaseCollection` and `BaseEntity` which you already imported and inherited; so no need of it.